# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: servo/ports/geckolib/glue.rs
# Commit: a16d763f2866
# Full Hash: a16d763f2866a1c44e67b85a66fa4503b5c764ac
# Author: Robert Longson <longsonr@gmail.com>
# Date: 2023-09-13 09:25:56
# Regressor Bug: 1287054
# File Overlap Count: 1
# Description:
#   Bug 1287054 part 1 - support rem, ch, ic and cap units for non-CSS lengths r=emilio,jgilbert
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D180618
# ==============================================================================

diff -r ccfee1091860 -r a16d763f2866 servo/ports/geckolib/glue.rs
--- a/servo/ports/geckolib/glue.rs	Tue Sep 12 20:57:49 2023 +0000
+++ b/servo/ports/geckolib/glue.rs	Tue Sep 12 21:10:54 2023 +0000
@@ -5352,6 +5352,18 @@
         structs::nsCSSUnit::eCSSUnit_XHeight => {
             NoCalcLength::FontRelative(FontRelativeLength::Ex(value))
         },
+        structs::nsCSSUnit::eCSSUnit_RootEM => {
+            NoCalcLength::FontRelative(FontRelativeLength::Rem(value))
+        },
+        structs::nsCSSUnit::eCSSUnit_Char => {
+            NoCalcLength::FontRelative(FontRelativeLength::Ch(value))
+        },
+        structs::nsCSSUnit::eCSSUnit_Ideographic => {
+            NoCalcLength::FontRelative(FontRelativeLength::Ic(value))
+        },
+        structs::nsCSSUnit::eCSSUnit_CapHeight => {
+            NoCalcLength::FontRelative(FontRelativeLength::Cap(value))
+        },
         structs::nsCSSUnit::eCSSUnit_Pixel => NoCalcLength::Absolute(AbsoluteLength::Px(value)),
         structs::nsCSSUnit::eCSSUnit_Inch => NoCalcLength::Absolute(AbsoluteLength::In(value)),
         structs::nsCSSUnit::eCSSUnit_Centimeter => {